AutoRest/AutoRest.Core/ClientModel/Parameter.cs

```diff
@@ -28,6 +28,20 @@ public Parameter()
         public string Name { get; set; }
 
         /// <summary>
+        /// Gets or sets the parameter client (explicitly defined code generation) name.
+        /// </summary>
+        public string ClientName {
```

Comment:
OK, I'll adjust it, but I'd appreciate your advice.

There's a central location to change the name of parameters in Autorest.Core.CodeNamer, which is why I placed ClientName on the model itself. If it should be in Extensions, we would essentially have to lift the logic to determine parameter names out of CodeNamer.cs, where it depends on the virtual method GetParameterName(). We would need to add a virtual method for the parameter naming in CodeNamer, and then pass the language-specific code namer in.

What if the CodeNamer code called the virtual parameter normalizing method? The choice would be language-specific, but not involve Extensions, which seems like a detour.